### PR TITLE
Add optional args to the toggle helper

### LINKED ISSUE #168: Toggle helper receives optional args of toggle values
E.g.

```hbs
<button {{action (toggle catSound this "meow" "nyan")}}>{{catSound}}</button>
<!-- <button>meow</button> -->
<!-- <button>nyan</button> -->
```

### PR DESCRIPTION
**Pull request by poteto**
Closes #168

### CODE REVIEW

File: addon/helpers/toggle.js [hunk 1]
```diff
@@ -1,10 +1,32 @@
...
+}
+
+export function toggle([prop, obj, ...values]) {
   return function() {
-    set(obj, prop, !get(obj, prop));
+    let currentValue = get(obj, prop);
+
+    if (isPresent(values)) {
+      let currentIdx = values.indexOf(currentValue);
+      let nextIdx = nextIndex(values.length, currentIdx);
```
Comment (martndemus) [inline addon/helpers/toggle.js]: `get(values, 'length')` to be safe.

Comment (poteto) [inline addon/helpers/toggle.js]: 👍

File: addon/helpers/toggle.js [hunk 2]
```diff
@@ -1,10 +1,32 @@
...
 import set from 'ember-metal/set';
+import { isPresent } from 'ember-utils';
 
-export function toggle([prop, obj]) {
+function nextIndex(length, currentIdx) {
+  let nextIdx;
+
+  if (currentIdx + 1 === length) {
+    nextIdx = 0;
+  } else if (currentIdx !== -1) {
```
Comment (martndemus) [inline addon/helpers/toggle.js]: Can you add a test that does not match either conditional above? (i.e. currentValue is not one of the toggle given states)

**Comment by martndemus** [comment]
Needs `addon/index.js` entry :grin:

**Comment by poteto** [comment]
This is an existing helper, it's already in `index`!

**Comment by martndemus** [comment]
Oops :grin:

**Pull request merged**